blog/nixos_remote_development/2023-11-16-Pilot.mdx

```diff
@@ -0,0 +1,199 @@
+---
+description: A minimal NixOS configuration to get started with remote development.
+slug: nixos-remote-development
+authors: shivaraj-bh
+tags: [nixos, minimal, flakes, nix]
+hide_table_of_contents: false
+---
+
+# NixOS Remote Development
+
+In this series of blog posts, we'll explore how to set up a remote development environment using NixOS. The goal of this post is to:
+- Create a minimal NixOS configuration.
+- Deploy the configuration (partition the disk and install the OS) to a remote machine over SSH (without a bootable USB).
+
+## Why?
+
+For starters, your system configuration is defined in code. This means that you don't have to login to see the state, like what users are present on the system, what services are running, etc. You can also deploy the configuration remotely.
+
+If you are someone who would like to use a powerful remote machine for development, while carrying around a lightweight laptop, then this post is for you.
+
+## The `flake.nix` file
+
+:::note
+If you are looking to just get started, you can skip this section and jump to the [next one](#deploying-the-configuration).
+:::
+
+:::note
+It is assumed that the user has a basic understanding of Nix (the language), if not, you can check out the rapid introduction to [Nix] and [Flakes].
+
+[Nix]: /nix-rapid
+[Flakes]: /nix-rapid#flakes
+:::
+
+```nix title="flake.nix"
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+    disko.url = "github:nix-community/disko";
+    disko.inputs.nixpkgs.follows = "nixpkgs";
+  };
+  outputs = { nixpkgs, disko, ... }:
+    {
+      nixosConfigurations.office = nixpkgs.lib.nixosSystem {
+        system = "x86_64-linux";
+        modules = [
+          disko.nixosModules.disko
+          ({ ... }: {
+            imports = [
+              ./disk-config.nix
+            ];
+            services.openssh.enable = true;
+            users.users = {
+              root = {
+                # Post-installation, the IP might change if MAC is not the
+                # only identifier used by DHCP server to lease an IP, by setting a
+                # password you can find the changed IP.
+                initialHashedPassword = "";
+                openssh.authorizedKeys.keys = [
+                  "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIFN5Ov2zDIG59/DaYKjT0sMWIY15er1DZCT9SIak07vK"
+                ];
+              };
+            };
+            boot.loader.grub = {
+              # adding devices is managed by disko
+              # devices = [ ];
+              efiSupport = true;
+              efiInstallAsRemovable = true;
+            };
+            system.stateVersion = "23.11";
+          }
+          )
+        ];
+      };
+    };
+}
+```
+
+### `nixosConfigurations` attribute
+
+It is a set of attributes, where each attribute is a NixOS configuration. This is how it looks like with two configurations:
```

Comment:
A blog post should have a smooth flow for a new reader who may not have prior knowledge or assumptions, so:

```suggestion
Flakes can output a special attribute called `nixosConfigurations` which can contain multiple NixOS configurations. It is a set of attributes, where each attribute is a NixOS configuration. This is how it looks like with two configurations:
```